### DISCUSSION TITLE
Building without Maven

### DISCUSSION THREAD

**Discussion by DiogoCarloto** [eclipse-milo/milo#1139] [Ideas]
I am using OPC UA in a JADE project on JDK 17, but this is not a Maven project. Can I add the Milo library to this project using a jar file?

**Comment by kevinherron** [comment]
I don't know what JADE is, but if you can gather up all the Milo JARs and all the dependency JARs then I don't see why not.

**Reply by DiogoCarloto** [reply]
Jade is a library for Multi Agent Systems, but is milo compatible with JDK 17, or It has to be built on JDK 8?

**Reply by kevinherron** [reply]
It is compiled with JDK 8, but it is runtime compatible with anything 8+.

If you are compiling it yourself you can use whatever JDK version you want. I think there's a broken security/certificate-related unit test when compiling with later JDKs, but it should work otherwise. I can't really see why you would be compiling the library yourself though.

**Reply by DiogoCarloto** [reply]
I appreciate your help; I am only trying to compile it myself because I have already finished and tested a "Java project" that I wanted to integrate with 4Diac using OPC UA. The problem is that this project is just a normal project, not a maven one.

**Reply by kevinherron** [reply]
I don't really see how compiling it yourself helps. Milo has a number of external dependencies. How are you going to get those on your class path as well?

**Reply by DiogoCarloto** [reply]
To be honest, I am just BrainStorming at the moment. Thank you; compiling from the source may be a bad idea. I will also look into mqtt, but I will probably have the same issue.
